src/NUnitFramework/framework/Assert.Exceptions.cs

```diff
@@ -41,25 +41,7 @@ public static Exception Throws(IResolveConstraint expression, TestDelegate code,
         {
             Exception caughtException = null;
 
-#if ASYNC
-            if (AsyncInvocationRegion.IsAsyncOperation(code))
-            {
-                using (var region = AsyncInvocationRegion.Create(code))
-                {
-                    code();
-
-                    try
-                    {
-                        region.WaitForPendingOperationsToComplete(null);
-                    }
-                    catch (Exception e)
-                    {
-                        caughtException = e;
-                    }
-                }
-            }
-            else
-#endif
+            Guard.ArgumentNotAsyncVoid(code, nameof(code));
```

Comment:
Shouldn't we be guarding that code shouldn't be `async` period?

Reply:
`code` is `TestDelegate` which is declared as `delegate void`, so if it's ever async, it's async void. I should add a comment to make that clear.

Reply:
`code` is `TestDelegate` which is declared as `delegate void`, so if it's ever async, it's async void. I added a comment to make that clear.